samples/SocialSample/Startup.cs

```diff
@@ -131,56 +133,61 @@ 127.0.0.1 MsSecSample.localhost.this
+                        response.EnsureSuccessStatusCode();
+
+                        var user = JObject.Parse(await response.Content.ReadAsStringAsync());
+                        
+                        var identifier = user.Value<string>("id");
+                        if (!string.IsNullOrEmpty(identifier))
                         {
-                            identity.AddClaim(new Claim(ClaimTypes.NameIdentifier, id, ClaimValueTypes.String, context.Options.ClaimsIssuer));
+                            notification.Identity.AddClaim(new Claim(
+                                ClaimTypes.NameIdentifier, identifier,
+                                ClaimValueTypes.String, notification.Options.ClaimsIssuer));
                         }
-                        var userName = user.TryGetValue("login", out value) ? value.ToString() : null;
+
+                        var userName = user.Value<string>("login");
                         if (!string.IsNullOrEmpty(userName))
                         {
-                            identity.AddClaim(new Claim(ClaimsIdentity.DefaultNameClaimType, userName, ClaimValueTypes.String, context.Options.ClaimsIssuer));
+                            notification.Identity.AddClaim(new Claim(
+                                ClaimsIdentity.DefaultNameClaimType, userName,
+                                ClaimValueTypes.String, notification.Options.ClaimsIssuer));
                         }
-                        var name = user.TryGetValue("name", out value) ? value.ToString() : null;
+
+                        var name = user.Value<string>("name");
                         if (!string.IsNullOrEmpty(name))
                         {
-                            identity.AddClaim(new Claim("urn:github:name", name, ClaimValueTypes.String, context.Options.ClaimsIssuer));
+                            notification.Identity.AddClaim(new Claim(
+                                "urn:github:name", name,
+                                ClaimValueTypes.String, notification.Options.ClaimsIssuer));
                         }
-                        var link = user.TryGetValue("url", out value) ? value.ToString() : null;
+
+                        var link = user.Value<string>("url");
                         if (!string.IsNullOrEmpty(link))
                         {
-                            identity.AddClaim(new Claim("urn:github:url", link, ClaimValueTypes.String, context.Options.ClaimsIssuer));
+                            notification.Identity.AddClaim(new Claim(
+                                "urn:github:url", link,
+                                ClaimValueTypes.String, notification.Options.ClaimsIssuer));
                         }
-
-                        context.Principal = new ClaimsPrincipal(identity);
                     },
                 };
             });
 
             // Choose an authentication type
-            app.Map("/login", signoutApp =>
```

Comment:
Definitely not something I wanted to change, but it sounds like @HaoK's recent changes broke something here: 

![image](https://cloud.githubusercontent.com/assets/6998306/8380151/8c417afe-1c24-11e5-9fa4-82796c3a3475.png)

It seems that the `redirect_uri` is now computed using the `PathBase` of the handler/middleware that triggers the `Challenge` call, which of course, produces an incorrect `redirect_uri` (`http://localhost:54540/login/signin-google` vs `http://localhost:54540/signin-google`).